sdgym/utils.py

```diff
@@ -74,29 +78,42 @@ def _get_synthesizer(synthesizer, name=None):
 
     if isinstance(synthesizer, str):
         if synthesizer.endswith('.json'):
+            LOGGER.info('Trying to load synthesizer from a JSON file.')
             with open(synthesizer, 'r') as json_file:
                 return json.load(json_file)
 
         baselines = Baseline.get_subclasses()
         if synthesizer in baselines:
+            LOGGER.info('Trying to import synthesizer by name.')
             synthesizer = baselines[synthesizer]
+
         else:
             try:
+                LOGGER.info('Trying to import synthesizer from fully qualified name.')
                 synthesizer = import_object(synthesizer)
+
             except Exception:
-                raise SDGymError(f'Unknown synthesizer {synthesizer}') from None
+                try:
+                    LOGGER.info('Trying to load synthesizer from JSON string.')
+                    return json.loads(synthesizer)
+
+                except JSONDecodeError as e:
```

Comment:
This try/except block is a bit strange, and I suspect that the `except Exception` part would never be reached, because at this point any string that is passed to the `json.loads` that is not a valid JSON will make it raise the `JSONDecodeError`.

Further than that, I would invert the order of things and do:
1. Try to load a JSON
2. If loading a JSON is not possible, try to import the synthesizer
3. If importing fails, it is an unknown synthesizer